[ES|QL] Theme updates for text based languages editor

ryankeairns opened the pull request:
## Summary

This is a design pass at the editor. Functionality and layout have not been changed.

### What has changed?
- Color settings in the `esql_theme.ts`
- Added additional rule groups for color treatment (e.g. more operators and values)
- Replaced hardcoded style values with EUI variables
- As a result, one set of theme colors now works across light and dark mode
- Made a few small Monaco configuration (i.e. options) changes that affect the UI/visual output (e.g. active line, font styles... the size is the same, but I believe these other changes will make it feel larger/more readable)
- Re-ordered the options set to more easily find what is/is not enabled
- Shortened the height of the top bar; tightened up the padding; more closely matches the footer design

### How does it look?
*Before-and-after comparison*
![CleanShot 2024-06-19 at 10 35 04@2x](https://github.com/elastic/kibana/assets/446285/54954b84-ae14-4a3f-b27b-d478dba43a33)

*Light mode*
![CleanShot 2024-06-19 at 10 29 26@2x](https://github.com/elastic/kibana/assets/446285/05d1ff28-e5db-422c-9961-d0743c589394)

*Dark mode*
![CleanShot 2024-06-19 at 10 28 27@2x](https://github.com/elastic/kibana/assets/446285/ecf10c8f-b461-4b71-928a-6b01ad72fe77)

*Compact*
![CleanShot 2024-06-17 at 16 15 26@2x](https://github.com/elastic/kibana/assets/446285/a0f2b129-5973-4742-b428-16ef4f8b1eca)

### What about the other things we saw in the Slack screenshots...
I have some other ideas cookin', but I would like to iterate on those concepts in Figma and get additional feedback. These entail the edge-to-edge design, less borders, consolidated header and footer, smaller CTA buttons, full-height field list, etc. Something like this...

<details>

<summary>Additional design considerations not in this PR</summary>

![CleanShot 2024-06-14 at 22 38 34@2x](https://github.com/elastic/kibana/assets/446285/dd5e0f12-1c9a-4ade-ad80-22376f841a41)

</details>

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-esql (Team:ESQL)

vadimkibana reviewed (approved):
Looks very nice!

---

One thing that was a bit unexpected to me is the pink autocomplete highlighted items. Maybe keep the Monaco default here?

<img width="448" alt="image" src="https://github.com/elastic/kibana/assets/82822460/873a72db-6599-4cb3-babe-6ab38dc0faa5">

ryankeairns commented:
> Looks very nice!
> 
> One thing that was a bit unexpected to me is the pink autocomplete highlighted items. Maybe keep the Monaco default here?
> 
> <img alt="image" width="448" src="https://private-user-images.githubusercontent.com/82822460/340488588-873a72db-6599-4cb3-babe-6ab38dc0faa5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2ODI5NTAsIm5iZiI6MTcxODY4MjY1MCwicGF0aCI6Ii84MjgyMjQ2MC8zNDA0ODg1ODgtODczYTcyZGItNjU5OS00Y2IzLWJhYmUtNmFiMzhkYzBmYWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE4VDAzNTA1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmMzhmNDhhMjNmZWU3NzllZDJiZTJlOGUxZDJmYzNlMDI5ZTQ0MjU1MzY0ZDY4NjI5NzgxMjNiYjg3ODk1OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.huJWKzsg2Cps0_O1HwJf014PEhhstBz5jDOH-EN0H7I">

Good catch! I meant to undo that.

stratoula reviewed (approved):
Sooooo beautiful I want to cry! I agree with Vadim that pink is too much but I dont like the monaco blue either. Maybe mint instead?

stratoula commented on packages/kbn-monaco/src/esql/lib/esql_theme.ts:
```diff
@@ -55,10 +46,16 @@ export const buildESQlTheme = (): monaco.editor.IStandaloneThemeData => ({
 
+    // source commands
+    ...buildRuleGroup(
+      ['from', 'row', 'show'],
```
yes lets add meta there too 👍

ryankeairns commented:
Updated suggestion styles
- EUI primary blue and rounded corners for current line
- Empty shade for suggest background
- Shadow on suggest panel

![CleanShot 2024-06-18 at 09 14 44@2x](https://github.com/elastic/kibana/assets/446285/34dc21d2-eb52-4e62-bb29-5a7b069e9b14)

![CleanShot 2024-06-18 at 09 51 42@2x](https://github.com/elastic/kibana/assets/446285/544bbb69-e645-4e1d-86d6-2bc18adb0e14)

...and couple bug fixes
- In compact mode, the appended buttons no longer move up a pixel on hover
- Removed extraneous border (actually a box shadow) on the editor/under the footer

timductive commented:
This dark mode is beautiful

andreadelrio reviewed (approved):
Such a great visual improvement! A couple of notes based on my review, no actions to be taken at the moment:

- I miss Roboto Mono a bit as I'm used to seeing code written in monospaced fonts. 
- As mentioned over Slack, I find it a bit confusing that some commands get bolded and some don't (e.g. COUNT) and I suspect the same might be true for other novice ES|QL users.

ryankeairns commented:
Added styles for the suggest detail (aka 'Read more') panel. Will address the typography separately.

![CleanShot 2024-06-18 at 14 47 07@2x](https://github.com/elastic/kibana/assets/446285/1abc30da-e7f2-4a7c-be6c-802c346cfca4)
